apigentools/utils.py

```diff
@@ -221,17 +232,17 @@ def write_full_spec(config, spec_dir, version, full_spec_file):
             else:
                 full_spec["paths"].update(loaded.get("paths", {}))
                 full_spec["tags"].extend(loaded.get("tags", []))
-                full_spec["components"]["schemas"].update(loaded.get("components", {}).get("schemas", {}))
-                full_spec["components"]["parameters"].update(loaded.get("components", {}).get("parameters", {}))
-                full_spec["components"]["securitySchemes"].update(loaded.get("components", {}).get("securitySchemes", {})),
-                full_spec["components"]["requestBodies"].update(loaded.get("components", {}).get("requestBodies", {})),
-                full_spec["components"]["responses"].update(loaded.get("components", {}).get("responses", {})),
-                full_spec["components"]["headers"].update(loaded.get("components", {}).get("headers", {})),
-                full_spec["components"]["examples"].update(loaded.get("components", {}).get("examples", {})),
-                full_spec["components"]["links"].update(loaded.get("components", {}).get("links", {})),
-                full_spec["components"]["callbacks"].update(loaded.get("components", {}).get("callbacks", {})),
                 full_spec["security"].extend(loaded.get("security", {}))
 
+                for field in COMPONENT_FIELDS:
+                    # Validate there aren't duplicate fields across files
+                    # Note: This won't raise an error if there is a duplicate component in a single file
+                    # That would alredy be deduped by the safe_load above.
+                    for key in loaded.get('components', {}).get(field, {}).keys():
+                        if key in full_spec.get('components', {}).get(field, {}):
+                            raise ValueError("Duplicate field {} found in spec. Exiting".format(key))
```

Comment:
It would be awesome if this printed which 2 yaml files contain this duplicity.

Reply:
At this point we have access to the file where we found the second occurrence , but we aren't keeping track of where we first found each field. While I agree seeing both file names would be useful, I'm not sure its worth the logic of keeping track of each field we see along the way. What do you think?

Reply:
Yeah, fair. We can always do it later.